Source/Matter/EMTensor.hpp

```diff
@@ -7,34 +7,64 @@
 #define EMTENSOR_HPP
 
 #include "CCZ4Geometry.hpp"
+#include "CCZ4RHSWithMatter.hpp"
 #include "Cell.hpp"
 #include "FourthOrderDerivatives.hpp"
-#include "Interval.H"
-#include "MatterCCZ4.hpp"
+#include "Interval.hpp"
+#include "VarsTools.hpp"
 #include "simd.hpp"
 
+// AMReX Includes
+#include <AMReX_MultiFab.H>
+
 //! Calculates the EM tensor and then saves the ones specified in the
 //! constructor on the grid
 template <class matter_t> class EMTensor
 {
   public:
     template <class data_t>
-    using Vars = typename MatterCCZ4<matter_t>::template Vars<data_t>;
+    using Vars = typename CCZ4RHSWithMatter<matter_t>::template Vars<data_t>;
+
+    /// derive record name
+    static inline const std::string name = "EMTensor";
+
+    /// Variable names
+    static inline amrex::Vector<std::string> var_names = {"rho"};
+
+    /// 3 components for the momentum density: Si_11, Si_22, Si_33
+    /// 6 components for the stress energy tensor (symmetric): Sij_11, Sij_12
+    /// etc.
+    static inline const amrex::Vector<std::string> extra_var_names = {
+        "Sij_11, Sij_22, Sij_33, Sij_11, Sij_12, Sij_13, Sij_22, Sij_23, "
+        "Sij_33"};
 
     //! Constructor
-    EMTensor(const matter_t &a_matter, const double dx, const int a_c_rho = -1,
+    EMTensor(const double dx, const int a_c_rho = -1,
              const Interval a_c_Si  = Interval(),
              const Interval a_c_Sij = Interval());
 
-    template <class data_t> void compute(Cell<data_t> current_cell) const;
+    template <class data_t>
+    AMREX_GPU_DEVICE AMREX_FORCE_INLINE void
+    compute(int i, int j, int k, const amrex::Array4<data_t> &out_mf,
+            const amrex::Array4<const data_t> &in_mf) const;
```

Comment:
Can we change these names to make it consistent with other compute classes
```suggestion
    compute(int i, int j, int k, const amrex::Array4<data_t> &out_arrays,
            const amrex::Array4<const data_t> &state_arrays) const;
```